### PR TITLE
Support non-uniform assemblies with `detailedAxialExpansion` disabled

### PR DESCRIPTION
**Pull request by onufer**
## Description

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/reactor/assemblies.py [hunk 1]
```diff
@@ -713,8 +713,13 @@ def setBlockMesh(self, blockMesh, conserveMassFlag=False, adjustList=None):
         belowFuelColumn = True
 
         if self[-1].p.topIndex == 0:
-            # this appears to not have been initialized, so initialize it
-            self.makeAxialSnapList(refMesh=blockMesh)
+            runLog.warning(
```
Comment (onufer) [inline armi/reactor/assemblies.py:716]: we initialize this in all assemblies, including blueprints assemblies in processLoading now, so every assembly should have this except for the ones we want to ignore expansion on.

**Review by jakehader — APPROVED** [review]
This is great! This will allow folks to implement certain assembly types (like control assemblies) in the blueprints without requiring them to strictly conform to the same mesh resolution as the reference assembly. This is useful in cases where assemblies can translate vertically throughout the core (like control rods). The main thing to look out for here is that physics solvers that require a uniform mesh must ensure that there is uniformity globally before running the analysis. This is implemented through specific physics plugins within said application.

**Comment by john-science** [comment]
Okay, this new `CaseSetting` seems fine, and I like the unit test.

But... who is planning on using this?  My thought is that since no one has needed this the last ~8 years, I'm surprised this went into `armi` and not a `Plugin` or `App`.  I assume that this was added because you want to use it yourself right now, right?

I'm kind of waffling; do we think this is generally useful enough to be in `ARMI`?

**Comment by onufer** [comment]
@john-science, we will likely be using within TP. please call me for a non-public chat i can expain.

**Pull request merged**